common/setup.py

```diff
@@ -29,9 +22,28 @@
     maintainer='Amundsen TSC',
     maintainer_email='amundsen-tsc@lists.lfai.foundation',
     packages=find_packages(exclude=['tests*']),
-    install_requires=requirements_common,
+    install_requires=[
+        # Packages in here should rarely be pinned. This is because these
+        # packages (at the specified version) are required for project
+        # consuming this library. By pinning to a specific version you are the
+        # number of projects that can consume this or forcing them to
+        # upgrade/downgrade any dependencies pinned here in their project.
+        #
+        # Generally packages listed here are pinned to a major version range.
+        #
+        # e.g.
+        # Python FooBar package for foobaring
+        # pyfoobar>=1.0, <2.0
+        #
+        # This will allow for any consuming projects to use this library as
+        # long as they have a version of pyfoobar equal to or greater than 1.x
+        # and less than 2.x installed.
+        'Flask>=1.0.2',
```

Comment:
Great point. Let me look into this. Would love to get rid of Flask here, or at the very least move it into an extra_requires.

Reply:
Sadly, although it's not used by amundsen itself, Flask is used to add logging. We use it internally at Lyft and I imagine so do others. It would be a backwards incompatible change to add it as an extra_requires. I am happy to do that and I agree it makes sense, but I suggest we take that change out separately.